### PR TITLE
ramips: add support for Asus RT-AC54U

### PR DESCRIPTION
**Pull request by p055877632**
ramips: add support for Asus RT-AC54U

### CODE REVIEW

**Comment by p055877632** [comment]
@Redfoxmoon3

File: target/linux/ramips/image/mt7620.mk [hunk 1]
```diff
@@ -362,6 +362,14 @@ define Device/rt-ac51u
 endef
 TARGET_DEVICES += rt-ac51u
 
+define Device/rt-ac54u
+  DTS := RT-AC54U
+  IMAGE_SIZE := $(ralink_default_fw_size_16M)
+  DEVICE_TITLE := Asus RT-AC54U
+  DEVICE_PACKAGES := kmod-usb-core kmod-usb2 kmod-usb-ehci kmod-usb-ledtrig-usbport
+endef
```
Comment (blogic) [inline target/linux/ramips/image/mt7620.mk]: kmod-usb-ehci does not exist, the ehci module is part of the usb2 package

File: target/linux/ramips/dts/RT-AC5XU.dtsi [hunk 2]
```diff
@@ -0,0 +1,122 @@
+/dts-v1/;
+
+#include "mt7620a.dtsi"
```
Comment (blogic) [inline target/linux/ramips/dts/RT-AC5XU.dtsi]: as you include the mt7620a.dtsi you can remove that line from the 2 dts files

**Comment by Redfoxmoon3** [comment]
I am not sure the SPI flash portion should go into the dtsi file, as the RT-AC52U and RT-AC53 should be identical in every other aspect. 52U has 128 MiB of NAND instead of SPI and the RT-AC53 has less SPI flash.

**Comment by p055877632** [comment]
@Redfoxmoon3 RT-AC53U CPU BCM5358UB0

AC51U AND AC54U  Twin models

AC52U  Large memory (FLA1: 128 MiB Zentel A5U1GA31ATS-BC
RAM1: 128 MiB Winbond W971GG6JB-25)

**Comment by Redfoxmoon3** [comment]
The RT-AC53U is a broadcom based router, yes. the RT-AC53 is MediaTek with 8 MiB of SPI instead of 16. We don't specify RAM in the DTS, so I don't see the problem there

**Comment by p055877632** [comment]
AC51U SPI ROM 16MB (128MiB)
AC52U SPI ROM 16MB (128MiB)
AC54U SPI ROM 16MB (128MiB) (MY ROUTER)

**Comment by Redfoxmoon3** [comment]
https://wikidevi.com/wiki/ASUS_RT-AC52U
https://wikidevi.com/wiki/ASUS_RT-AC51U

AC52U has 128 MiB NAND :-)

**Comment by p055877632** [comment]
FIX FINAN

**Comment by blogic** [comment]
please make sure to squash all your commits into one

**Comment by p055877632** [comment]
file is ready.....

**Comment by Redfoxmoon3** [comment]
Please follow the guide here for squashing commits. https://lede-project.org/submitting-patches#squashing_commits

**Comment by p055877632** [comment]
squash done

File: target/linux/ramips/base-files/etc/board.d/01_leds [hunk 3]
```diff
@@ -321,6 +321,10 @@ rt-ac51u)
 	set_wifi_led "$board:blue:wifi"
 	set_usb_led "$board:blue:usb" "1-1"
 	;;
+rt-ac54u)
```
Comment (heinzek) [inline target/linux/ramips/base-files/etc/board.d/01_leds]: look up to section rt-51u. its the same as rt-ac54u.

File: target/linux/ramips/dts/RT-AC51U.dts [hunk 4]
```diff
@@ -116,6 +116,20 @@
 	ralink,mtd-eeprom = <&factory 0>;
 };
 
+&pcie {
```
Comment (mkresin) [inline target/linux/ramips/dts/RT-AC51U.dts]: Please drop this change from your PR! It is unrelated to "add support for Asus RT-AC54U".

@Redfoxmoon3 Is this change required at all for the AC51U?

Comment (Redfoxmoon3) [inline target/linux/ramips/dts/RT-AC51U.dts]: It will be if the 5 GHz chip is supported in the future.

**Comment by mkresin** [comment]
@p055877632 Do you plan to fix the outstanding issues?

Your PR still has the RT-AC51U changes and your commit message needs to be updated with hardware description, LEDE installation howto and your Signed-off-by line. Please have a look at our [patch submitting guideline](https://lede-project.org/submitting-patches#submitting_patches).

**Comment by mkresin** [comment]
Closed due to inactivity. Please reopen the pull request if you have the time to fix the outlined issues.
